transforms/dec128.js

```diff
@@ -0,0 +1,78 @@
+const DECIMAL_LITERAL = "DecimalLiteral";
+
+const opToName = {
+  '+': 'add',
+  '*': 'mul',
+  '-': 'sub',
+  '/': 'div',
+};
+
+const includedOps = Object.keys(opToName);
+
+const isDecimalOrBinaryExpBare = (t) => (node) => {
+  return isDecimal(node) || t.isBinaryExpression(node);
+};
+
+const areMixedTypesBare = (isDecimalOrBinaryExp) => (a, b) => {
+
+  return (isDecimalOrBinaryExp(a) && !isDecimalOrBinaryExp(b))
+    || !(isDecimalOrBinaryExp(a) && isDecimalOrBinaryExp(b))
+};
+
+const isDecimal = (node) => {
+  return node.get('callee').isIdentifier({name: 'Decimal'}) || node.type === DECIMAL_LITERAL;
+};
+
+const replaceWithDecimalExpression = (t) => (path) => {
+  const isDecimalOrBinaryExp = isDecimalOrBinaryExpBare(t);
+  const areMixedTypes = areMixedTypesBare(isDecimalOrBinaryExp);
+
+  const { left, right, operator } = path.node;
+
+  if (!isDecimalOrBinaryExp(path.get('left')) && !(isDecimalOrBinaryExp(path.get('right')))) {
+    return;
+  }
+
+  if (areMixedTypes(path.get('left'), path.get('right'))) {
+    throw new Error ('Mixed numeric types are not allowed.');
+  }
+
+  if (!includedOps.includes(operator)) {
+    throw new Error (`${operator} is not currently supported.`);
```

Comment:
You can throw the error using `throw path.buildCodeFrameError("error messge")`, so that the error points out the location of the problem.